flowey/flowey_lib_hvlite/src/install_openvmm_rust_build_essential.rs

```diff
@@ -33,6 +34,10 @@ impl FlowNode for Node {
             ctx.reqv(crate::init_openvmm_cargo_config_deny_warnings::Request::Done),
             ctx.reqv(crate::init_openvmm_magicpath_protoc::Request),
             ctx.reqv(flowey_lib_common::install_rust::Request::EnsureInstalled),
+            ctx.reqv(|v| flowey_lib_common::install_apt_pkg::Request::Install {
+                package_names: vec!["build-essential".into()],
```

Comment:
maybe a hot take, but I strongly think `build-essential` is an anti-pattern, at least in the context of flowey.

its fine for local dev, where you probably want to "go fast, and get things working", but if you do the same thing in CI, you'll either waste time installing things you don't need, or end up installing dependencies that things will take ambient deps on (and then have those things break if/when they move to a different pipeline that _doesn't_ have an ambient build-essential install)

* * *

maybe a more meta observation is that this node might have simply outlived its usefulness. it has been around from the _very_ early days of flowey development, and I think we have a far better toolkit available to do what this node is trying to do:

1. `init_openvmm_cargo_config_deny_warnings` should likely be something which should be a dep of "git_checkout_hvlite_repo", either directly, or via a patch
2. `init_openvmm_magicpath_protoc` and  `flowey_lib_common::install_rust` should likely just be direct deps of the various `flowey_lib_hvlite::cargo_*` nodes, and any other nodes which invoke cargo directly on crates that need it

or maybe we do keep it around, but change its name to better reflect what it is trying to do?

or maybe it gets "enlightened" to the particular target-triple that needs to get set up, and it itself invokes the `init_cross_build` node?

idk, lots of different ways to model this

Reply:
I agree with you that we should try to install only the bare minimum dependencies, but I think it is going to take a lot of work to untangle all the deps in flowey and achieve that. I was just trying to move a few things around to make it make more sense to me (I don't think any of this reshuffling meaningfully changed any of the yaml). For this PR I'll just change it back to the way it was before, and we can save this refactor for subsequent PR(s).